component/azstorage/config.go

```diff
@@ -390,10 +390,23 @@ func ParseAndValidateConfig(az *AzStorage, opt AzStorageOptions) error {
 		az.stConfig.authConfig.SASKey = sanitizeSASKey(opt.SaSKey)
 	case EAuthType.MSI():
 		az.stConfig.authConfig.AuthMode = EAuthType.MSI()
-		if opt.ApplicationID == "" && opt.ResourceID == "" {
+		if opt.ApplicationID == "" && opt.ResourceID == "" && opt.ObjectID == "" {
 			//lint:ignore ST1005 ignore
 			return errors.New("Application ID and Resource ID not provided")
 		}
+		v := make(map[string]bool, 3)
```

Comment:
if opt.ApplicationID == "" && opt.ResourceID == "" && opt.ObjectID == "" {
this check is already validating the same so I think this is just duplicate check and can be removed.

Reply:
no that check is validating if nothing got passed and we're type msi the other is validating if we passed more than 1 type of msi credential. I will combine both but they do different checks

Reply:
did not get that, here in if condition we were checking either of them shall be present. In the below code where we are adding items to map, we are checking exactly one of them shall be there then they both are the same. If so, if condition is a better check, then constructing a map and add values.
If condition is expected to check multiple values can be provided out of these three things, then also if condition will pass and eventually, we will end up passing down all three to the sdk and it will decide which one to prioritize.
Not able to understand the difference between if and the map length checks.

Reply:
I think Vikas's comment makes sense, instead of creating a map, we can just convert this check into some if statements

Reply:
I fail to see the point still. why is multiple if conditions (since we can't XOR here) better than the map? I actually think the if condition(s) will be not as readable.

Reply:
you could create a function that takes in 3 strings as input and returns whether only the first is set so like a != "" && b == ""  && c== "" and then call the function like this (func(a, b, c) || func(b, a, c) || func(c, a, b))

Reply:
why is that better than a map